payment/src/main/java/org/killbill/billing/payment/core/ProcessorBase.java

```diff
@@ -134,6 +136,15 @@ protected PaymentPluginApi getPaymentProviderPlugin(final UUID paymentMethodId,
         return getPaymentPluginApi(methodDao.getPluginName());
     }
 
+    protected String getPaymentProviderPluginName(final UUID paymentMethodId, final InternalTenantContext context) throws PaymentApiException {
+        final PaymentMethodModelDao methodDao = paymentDao.getPaymentMethodIncludedDeleted(paymentMethodId, context);
+        if (methodDao == null) {
+            log.error("PaymentMethod does not exist", paymentMethodId);
```

Comment:
I am a nit concerned about this extra DAO call just to retrieve the plugin name? Can't we access that info without incurring the cost of an extra query?